content/articles/searching-and-sorting-techniques/index.md

```diff
@@ -0,0 +1,177 @@
+#### 4).Merge Sort
+One of the most efficient sorting algorithms is merge sort. It is based on the divide-and-conquer strategy. Merge sort reduces down a list into sublists until every sublist only includes one element, after which it combines the sublists into a sorted list.
+##### Algorithm
+step 1: If there is just one entry in the list return it because it has been sorted already.
+
+Step 2:  Recursively partition the list into two halves until you can't partition it anymore.
+
+Step 3: Sort the smaller lists and combine them to make a new list.
+#### 5. Quick Sort
+It is a split-and-conquer strategy, similar to Merge Sort. It selects a pivot element and partitions the specified array around that pivot.
+
+A large list is partitioned into two sublists, the first of which includes elements lower than the given element, referred to as pivots, and the other of which contains elements higher than the pivot element, on which divisions are constructed.
+##### Algorithm
+Step 1: Decide which index value has the highest pivot.
+
+Step 2: Use two parameters, one is on the upper left of the list middle value and the other on the upper right.
+
+Step 3 to the left indicates the low index.
+
+Step 4: The right arrow points to the highest point.
+
+Step 5: While the element on the left is lower than the pivot, go right.
+
+Step 6: move left while the value at the right is bigger than the pivot.
+
+Step 7: Swap left and right if both steps 5 and 6 don't match.
+
+Step 8: If they meet on the left or right, the place where they met becomes the new pivot.
+#### 6. Heap Sort
+Heapsort is a sorting algorithm that compares objects using a Binary Heap dataset. It's comparable to selection sorting, in which we identify the smallest piece first and place it at the top. For the remaining values, the same procedure is repeated.
+
+A Binary Heap is a Complete Binary Tree in which elements are placed in such a way that the value of a parent node is higher (or lower) than the elements of its two offspring nodes.
+##### Algorithm
+Sorting in ascending order with the Heap Sort Algorithm:
+
+step 1: Create a maximum heap using the given data.
+
+step 2: At this point, the largest item is placed at the base of the heap.
+
+step 3:Replace it with the heap's final item, then reduce the heap's size by one.
+ 
+step 4:Finally, heapify the tree's root.
+
+step 5:Repeat steps 2,3 and 4 as long as the heap is larger than one.
+### Conclusion
+Finding a specified element in an array of ‘t' elements is referred to as searching in data structures. In terms of searching, there are two kinds of searches: sequential and interval. Almost every search algorithm falls into one of these two types. 
+
+
+Binary and linear searches are two quick and easy techniques, with binary working quicker than linear search techniques.
+
+Binary search is quicker if the acquired data is sorted and the array size is large.
+

```

Comment:
Add a `Further reading` section where you add external links that helps reader gain more knowledge on this topic